moveit_core/trajectory_processing/include/moveit/trajectory_processing/time_optimal_trajectory_generation.h

```diff
@@ -174,13 +174,30 @@ class TimeOptimalTrajectoryGeneration : public TimeParameterization
                          const std::unordered_map<std::string, double>& velocity_limits,
                          const std::unordered_map<std::string, double>& acceleration_limits) const override;
 
+  // clang-format off
+  /**
+   * \brief Compute a trajectory with the given number of waypoints (plus or minus 1, due to rounding).
+   * Resampling the trajectory to get the desired num_waypoints doesn't change the start and goal point,
+   * and all re-sampled waypoints will be on the path of the original trajectory (within the configured tolerances)
+   * but controller execution may deviate from the intended path if waypoint spacing is too sparse.
+   * \param num_waypoints The desired number of waypoints.
+   * \param[in,out] trajectory A path which needs time-parameterization. It's OK if this path has already been
+   * time-parameterized; this function will re-time-parameterize it.
+   * \param max_velocity_scaling_factor Joint velocity limits are scaled by this factor.
+   * \param max_acceleration_scaling_factor Joint acceleration limits are scaled by this factor.
+   */
+  // clang-format on
+  bool computeTimeStamps(const size_t num_waypoints, robot_trajectory::RobotTrajectory& trajectory,
+                         const double max_velocity_scaling_factor = 1.0,
+                         const double max_acceleration_scaling_factor = 1.0);
+
 private:
   bool doTimeParameterizationCalculations(robot_trajectory::RobotTrajectory& trajectory,
                                           const Eigen::VectorXd& max_velocity,
                                           const Eigen::VectorXd& max_acceleration) const;
 
   const double path_tolerance_;
-  const double resample_dt_;
+  double resample_dt_;
```

Comment:
@AndyZe it's actually error-prone that `resample_dt_` is being changed implicitly by the new `computeTimeStamps()` function. The other member functions have been `const` for a reason, namely to allow repeatable results. Now the results of the old `computeTimeStamps()` depends on if the new function has been called and whatever `num_waypoints` has been used. Please fix that, either by implementing a free function that creates two instances of TOTG (one for each resample_dt), or by providing a private `computeTimeStampsInternal` function that accepts `resample_dt` as argument so that you can call that one twice without changing the member variable.

Reply:
OK, I'll fix this soon